---
title: "SoupX_PreProcess"
author: "Drake Williams"
date: "`r BiocStyle::doc_date()`"
output: html_document
knit: (function(inputFile, encoding) { 
    rmarkdown::render(
        inputFile, encoding = encoding, 
        output_file = file.path(
            dirname(inputFile), paste0('01_Soupx_',Sys.Date(),'.html'))) 
                })
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      warning = F,
                      message = F, 
                      out.width = "100%", 
                      fig.align='center')
options(width = 1200)
```

# Introduction

The objective of this notebook is to estimate and remove cell free mRNA contamination in droplet based single cell RNA-seq data and generate Seurat objects for further analysis

## Load packages

```{r load packages}
library(pacman)
p_load(Seurat, SoupX, scCustomize, stringr)
```

## Desoup datasets
```{r desoup data}
scCustomize::Setup_scRNAseq_Project(custom_dir_file = paste0(getwd(), "/scSeq_dirs.csv"))
source("~/updatedAtlas-hu/01_scripts/soupxHelpers.R")

# get the full directory list (recursively) to pick out folders with data to be analyzed
dirList <- list.dirs(path = "~/updatedAtlas-hu/02_raw_data", full.names = TRUE, recursive = TRUE)
# a list of strings that will keep all of the folder paths that contain sequencing information
# main purpose is to get rid of higher level folders
# case matters
dirList <- dirList[grepl("GRCh38|raw_feature_bc_matrix|filtered_feature_bc_matrix", dirList)]
# remove folders that contain samples are not relevant to the current analysis
# case matters
dirList <- dirList[!grepl("Col2|GSE",dirList)]
health_dirs <- dirList[grepl("ging_hlth|bucl_hlth", dirList)]
perio_dirs <- dirList[grepl("ging_prio", dirList)]
id_dirs <- dirList[grepl("ging_lad1|ging_stat", dirList)]
id_dirs <- id_dirs[!grepl("poorQuality|ID080/", id_dirs)]


print("These are the directories that will be used for desouping. Double check for errors.")
print(dirList)
print(health_dirs)
print(perio_dirs)
print(id_dirs)

healthDesouped <- desoupGroup(health_dirs)

perioDesouped <- desoupGroup(perio_dirs)

idDesouped <- desoupGroup(id_dirs)
```

## Generate seurat objects from dgCMatrices
```{r generate seurat lists}
# Make seurat objects from desouped dgCMatrices

healthSeurat <- dcgToSeurat(healthDesouped,
                            health_dirs)


perioSeurat <- dcgToSeurat(perioDesouped,
                           perio_dirs)

idSeurat <- dcgToSeurat(idDesouped,
                         id_dirs)

```

## Merge and save datasets
```{r merge seurat objects}
baseDir <- "~/updatedAtlas-hu/04_data_objects/01_SoupX_objects/"
# The cell IDs become 'orig.ident' in the metadata
all <- Merge_Seurat_List(list_seurat = c(healthSeurat,perioSeurat,idSeurat))
saveRDS(all, file = paste0(baseDir, Sys.Date(), "_01-SoupxOutput.RDS"))

```

## Session Info
```{r session info}
sessionInfo()
```

